python/simulation/fake_raft.py

```diff
@@ -0,0 +1,637 @@
+    test_type = kwargs.get('test_type', None)
+
+    try:
+        folder = os.environ['LCATR_DATACATALOG_FOLDER']
+    except KeyError:
+        folder = make_datacat_path(root_folder=root_folder, sensor_type=sensor_type,
+                                   sensor_id=sensor_id)
+
+    datacat = DataCatalog(folder=folder, site=kwargs.get('site', 'slac.lca.archive'))
+    query = '&&'.join(('LSST_NUM=="%s"' % sensor_id,
+                       'ProcessName=="%s"' % process_name))
+    if image_type is not None:
+        query += '&& IMGTYPE == "%s"' % image_type
+    if test_type is not None:
+        query += '&& TESTTYPE == "%s"' % test_type
+
+    file_list = []
+    try:
+        datasets = datacat.find_datasets(query)
+    except DcClientException as eobj:
+        # Make the error message a bit more useful for debbuging
+        msg = eobj.message + (":\nFolder = %s\n" % folder)
+        msg += "Query = %s\n" % query
+        sys.stderr.write(msg)
+        return file_list
+        #raise DcClientException(msg)
+
+    for item in datasets.full_paths():
+        if fnmatch.fnmatch(os.path.basename(item), pattern):
+            file_list.append(item)
+
+    file_list = sort_unique(file_list)
+    if kwargs.get('sort', False):
+        file_list = sorted(file_list)
+
+    return file_list
+
+
+class RaftImages(object):
+    '''
+    Writes a raft's worth of images based on a user-supplied single-sensor
+    image.
+
+    Parameters
+    ----------
+    raft_id : str
+        Name of the raft, e.g., 'RAFT_000'.  This is used to evaluate coordinate
+        keywords for focal plane coordinates
+    process_name : str
+        Name of the 'test' being applied.  This probably could be extracted
+        from single_sensor_file.  No analysis is implied; the string is just
+        used for assigning names to output files.
+    sensor_type: str
+        ITL-CCD or E2V-CCD
+    output_path : str
+        Path to prepended to the output file names
+
+    Attributes
+    ----------
+    ccd_image : astropy.io.fits.HDUList
+        This is an Astropy HDUList that contains all the headers and image
+        extensions of the single_sensor_file
+    '''
+
+    def __init__(self, raft_id, process_name, sensor_type, output_path):
+        """
+        Class constructor.
+        """
+        self.raft_id = raft_id
+        self.process_name = process_name
+        self.sensor_type = sensor_type
+        self.output_path = output_path
+        try:
+            os.mkdir(self.output_path)
+        except OSError:
+            pass
+
+    def update_primary_header(self, slot_name, hdu):
+        """
+        Update the primary header
+
+        Parameters
+        ----------
+        slot_name : str
+            Name of the slot within the raft
+        hdu : fits.Image
+            FITS image whose header is being updated
+        """
+        print ("Placeholder", self.raft_id, slot_name, hdu)
+
+    def update_image_header(self, slot_name, ext_num, hdu):
```

Comment:
Good point about self.  update_image_header does not need to know where the raft is in the focal plane in order to add the raft-level coordinates.  It would need to know the slot in the focal plane in order to add focal plane-level coordinates.  Because the directory path convention has just the <raft ID> and not the location of the raft in the focal plane, in order to eventually add focal plane coordinates we'll need to make up a mapping between raft ID and focal plane location, I'd guess something like an extension of test_raft.yaml.  And update_image_header could get the raft ID from self.